### PR TITLE
ARROW-17319: [Python] Update VCPKG to include updated aws-c-io

### PR DESCRIPTION
**Pull request by mgeplf**
Should solve https://issues.apache.org/jira/browse/ARROW-17319

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -93,7 +93,7 @@ DEVTOOLSET_VERSION=-1
 # Please also update the crossbow configuration in order to keep the github
 # actions cache up to date for the macOS wheels:
 #   https://github.com/ursacomputing/crossbow/blob/master/.github/workflows/cache_vcpkg.yml
-VCPKG="38bb87c"
+VCPKG="7ef97003d78178948068ccd6fa7964f938f41c76"
```
Comment (kou) [inline .env:96]: Please use short SHA as the above comment says:

```suggestion
VCPKG="7ef9700"
```

**Comment by kou** [comment]
@github-actions crossbow submit -g wheel

**Comment by kou** [comment]
@github-actions crossbow submit -g wheel

**Comment by amol-** [comment]
Closing because it has been untouched for a while, in case it's still relevant feel free to reopen and move it forward 👍

**Pull request closed without merging**